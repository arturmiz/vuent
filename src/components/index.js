import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
