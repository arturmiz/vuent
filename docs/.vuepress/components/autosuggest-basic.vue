<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
