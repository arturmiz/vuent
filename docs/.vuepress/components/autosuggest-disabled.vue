<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
