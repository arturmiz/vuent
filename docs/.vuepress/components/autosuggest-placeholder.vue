<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
