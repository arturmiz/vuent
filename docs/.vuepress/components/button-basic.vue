<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
