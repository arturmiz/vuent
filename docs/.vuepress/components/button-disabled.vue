<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
