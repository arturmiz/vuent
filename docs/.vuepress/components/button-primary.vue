<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
