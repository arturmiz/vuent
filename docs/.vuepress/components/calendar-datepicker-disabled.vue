<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
