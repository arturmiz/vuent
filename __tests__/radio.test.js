import { createLocalVue, mount } from '@vue/test-utils';
import { VntRadio } from '@/components';
import { isInstalled } from './utils';

describe('RadioButton', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntRadio);

    expect(isInstalled(localVue, VntRadio)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntRadio);
    });

    test('is not disabled', () => {
      expect(wrapper.vm.disabled).toBe(false);
    });

    test('is not checked', () => {
      expect(wrapper.attributes().checked).toBe(undefined);
    });

    test('name is "radio"', () => {
      expect(wrapper.vm.name).toBe('radio');
    });

    test('label text is "Radio"', () => {
      expect(wrapper.vm.label).toBe('Radio');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can be disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntRadio, {
        propsData: {
          disabled: true
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

  });

  describe('can be checked', () => {

    beforeAll(() => {
      wrapper = mount(VntRadio, {
        attrs: {
          checked: 'true'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can be checked and disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntRadio, {
        attrs: {
          checked: 'true'
        },
        propsData: {
          disabled: true
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

  });

  test('can have content set using slot', () => {
    wrapper = mount(VntRadio, {
      slots: {
        default: '<span>Custom content</span>'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('can have name attribute', () => {

    beforeAll(() => {
      wrapper = mount(VntRadio, {
        propsData: {
          name: 'groupedRadios'
        }
      });
    });

    test('set correctly', () => {
      expect(wrapper.vm.name).toBe('groupedRadios');
    });

    test('rendered correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('when not disabled, emits selected value', () => {

    beforeAll(() => {
      wrapper = mount(VntRadio, {
        attrs: {
          value: 'red'
        },
        propsData: {
          disabled: false
        }
      });

      const label = wrapper.find('label');
      label.trigger('click');
    });

    test('event is emitted', () => {
      expect(wrapper.emitted()['input']).toBeDefined();
    });

    test('new checked state is emitted', () => {
      expect(wrapper.emitted()['input'][0]).toEqual(['red']);
    });

  });

  describe('when disabled, doesn\'t emit selected value', () => {

    beforeAll(() => {
      wrapper = mount(VntRadio, {
        attrs: {
          value: 'blue'
        },
        propsData: {
          disabled: true
        }
      });
    });

    test('event is not emitted', () => {
      const label = wrapper.find('label');
      label.trigger('click');

      expect(wrapper.emitted()['input']).toBeUndefined();
    });

    test('toggle doesn\'t emit event', () => {
      const toggleResult = wrapper.vm.toggle({
        target: { value: 'blue' }
      });
      expect(toggleResult).toBeUndefined();
      expect(wrapper.emitted()['input']).toBeUndefined();
    });

  });

});
