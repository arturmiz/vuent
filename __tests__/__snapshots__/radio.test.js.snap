// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton by default renders correctly 1`] = `
<label class="vnt-radio">
  <input type="radio" name="radio" class="vnt-radio__input"> <span class="vnt-radio__icon"></span> <span class="vnt-radio__text">Radio</span></label>
`;

exports[`RadioButton can be checked and disabled renders correctly 1`] = `
<label class="vnt-radio" checked="checked">
  <input type="radio" disabled="disabled" name="radio" class="vnt-radio__input"> <span class="vnt-radio__icon"></span> <span class="vnt-radio__text">Radio</span></label>
`;

exports[`RadioButton can be checked renders correctly 1`] = `
<label class="vnt-radio" checked="checked">
  <input type="radio" name="radio" class="vnt-radio__input"> <span class="vnt-radio__icon"></span> <span class="vnt-radio__text">Radio</span></label>
`;

exports[`RadioButton can be disabled renders correctly 1`] = `
<label class="vnt-radio">
  <input type="radio" disabled="disabled" name="radio" class="vnt-radio__input"> <span class="vnt-radio__icon"></span> <span class="vnt-radio__text">Radio</span></label>
`;

exports[`RadioButton can have content set using slot 1`] = `
<label class="vnt-radio">
  <input type="radio" name="radio" class="vnt-radio__input"> <span class="vnt-radio__icon"></span> <span class="vnt-radio__text"><span>Custom content</span></span>
</label>
`;

exports[`RadioButton can have name attribute rendered correctly 1`] = `
<label class="vnt-radio">
  <input type="radio" name="groupedRadios" class="vnt-radio__input"> <span class="vnt-radio__icon"></span> <span class="vnt-radio__text">Radio</span></label>
`;
